core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -54,15 +54,19 @@ public Sep10Service(
     Log.info("Sep10Service initialized.");
   }
 
-  public ChallengeResponse createChallenge(ChallengeRequest challengeRequest) throws SepException {
+  public ChallengeResponse createChallenge(ChallengeRequest challengeRequest)
+      throws SepException, MalformedURLException {
     info("Creating challenge");
     //
     // Validations
     //
     if (challengeRequest.getHomeDomain() == null) {
-      debugF("home_domain is not specified. Will use the default: {}", sep10Config.getHomeDomain());
-      challengeRequest.setHomeDomain(sep10Config.getHomeDomain());
-    } else if (!sep10Config.getHomeDomain().equalsIgnoreCase(challengeRequest.getHomeDomain())) {
+      debugF(
+          "home_domain is not specified. Will use the default: {}", sep10Config.getWebAuthDomain());
+      challengeRequest.setHomeDomain(sep10Config.getWebAuthDomain());
```

Comment:
We're not using the correct default here, it should be `appConfig.getHostUrl()`.
```suggestion
      debugF(
          "home_domain is not specified. Will use the default: {}", getDomainFromURL(appConfig.getHostUrl()));
      challengeRequest.setHomeDomain(getDomainFromURL(appConfig.getHostUrl()));
```

Reply:
Fixed.